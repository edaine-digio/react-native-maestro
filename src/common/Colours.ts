export const Colours = {
  Black: '#000',
  White: '#FFF',
  Sand: '#FFD1AD',
  Flax: '#F5EDE0',
  Peach: '#FC9468',
  Denim: '#243746',
  LightGrey: '#D1D1D1',
  MedGrey: '#8A8A8A',
  LightButter: '#FFFAED',
  Butter: '#FFC844',
  LightFlamingo: '#FFF3F4',
  Flamingo: '#FF8189',
  LightSpring: '#E6FBF2',
  Spring: '#00CE7C',
  LightViolet: '#F3EFFA',
  Violet: '#7E57C5',
  Transparent: 'transparent'
}
