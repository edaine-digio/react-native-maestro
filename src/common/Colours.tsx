export const Colours = {
  Black: '#000',
  White: '#fff',
  Spring: '#00CE7C',
  Violet: '#7E57C5',
  Flamingo: '#FF8189',
  Butter: '#FFC844',
  Denim: '#243746',
  Flax: '#F5EDE0',
};
