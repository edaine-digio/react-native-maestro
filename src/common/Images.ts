import DigioPrimarySvg from 'src/assets/icons/DigioPrimary.svg'
import DigioIconSvg from 'src/assets/icons/DigioIcon.svg'
import ChevronLeftSvg from 'src/assets/icons/ChevronLeft.svg'
import HomeTabIconSvg from 'src/assets/icons/HomeTabIcon.svg'
import ProfileTabIconSvg from 'src/assets/icons/ProfileTabIcon.svg'
import SettingsTabIconSvg from 'src/assets/icons/SettingsTabIcon.svg'
import SendIconSvg from 'src/assets/icons/SendIcon.svg'

export const SvgImages = {
  DigioPrimary: DigioPrimarySvg,
  DigioIcon: DigioIconSvg,
  ChevronLeft: ChevronLeftSvg,
  HomeTabIcon: HomeTabIconSvg,
  ProfileTabIcon: ProfileTabIconSvg,
  SettingsTabIcon: SettingsTabIconSvg,
  SendIcon: SendIconSvg
}
