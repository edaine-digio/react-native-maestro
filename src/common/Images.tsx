export const Images = {
  appLogo: require('assets/appLogo.png'),
  digio: require('assets/Digio.png'),
};
