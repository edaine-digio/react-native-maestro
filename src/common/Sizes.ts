export const Padding = {
  XXS: 2,
  XS: 4,
  SM: 8,
  MD: 16,
  LG: 24,
  XL: 32,
  XXL: 48
}

export const Font = {
  XXS: 8,
  XS: 12,
  SM: 16,
  MD: 20,
  LG: 24,
  XL: 32,
  XXL: 48
}
