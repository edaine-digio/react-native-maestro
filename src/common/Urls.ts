import { Platform } from 'react-native'

export const BASE_URL =
  Platform.OS === 'ios' ? 'http://localhost:4001' : 'http://10.0.2.2:4001'
