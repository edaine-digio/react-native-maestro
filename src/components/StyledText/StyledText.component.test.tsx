import React from 'react'
import { render, screen } from '@testing-library/react-native'
import { StyledText, TextAlign } from './StyledText.component'
import { Colours } from 'src/common/Colours'

describe('StyledText', () => {
  test('Should render correctly with default values', () => {
    render(<StyledText fontSize={18}>This is a test</StyledText>)

    expect(screen.toJSON()).toMatchSnapshot()
  })

  test('Should render correctly with custom values', () => {
    render(
      <StyledText
        fontSize={40}
        fontWeight={800}
        color={Colours.Sand}
        alignment={TextAlign.Right}>
        This is a test
      </StyledText>
    )

    expect(screen.toJSON()).toMatchSnapshot()
  })
})
