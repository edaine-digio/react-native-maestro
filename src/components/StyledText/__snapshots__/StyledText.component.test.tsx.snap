// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText Should render correctly with custom values 1`] = `
<Text
  style={
    {
      "color": "#FFD1AD",
      "fontFamily": "Archivo-ExtraBold",
      "fontSize": 40,
      "textAlign": "right",
    }
  }
>
  This is a test
</Text>
`;

exports[`StyledText Should render correctly with default values 1`] = `
<Text
  style={
    {
      "color": "#000",
      "fontFamily": "Archivo-Regular",
      "fontSize": 18,
      "textAlign": "right",
    }
  }
>
  This is a test
</Text>
`;
